### ISSUE TITLE
New-SFSchedule fails when using -Retention* parameters

### ISSUE THREAD

**Issue by rpmcdougall** [solidfire/PowerShell#35]
When using the New-SFSchedule cmdlet I was able to deduce that when I set any sort of retention on a snapshot, it would fail with the same error as noted below. I also tried using the Set-SFSchedule cmdlet to set it after the schedule was created with out retention and received an identical error.  Let me know if there is anything I might be doing wrong or if you need any additional information for me to properly report his issue.

**Steps to reproduce below:**
**This is with retention set...**
```

 New-SFSchedule -VolumeID $test.VolumeID -ScheduleName "TEST-SF01-T2-TEST99" -Frequency Time
Interval -TimeIntervalHours 6 -SnapshotName "Every6hFor24h-TEST-SF01-T2-TEST99" -RetentionHours 24
New-SFSchedule : Nullable object must have a value.
At line:1 char:1
+ New-SFSchedule -VolumeID $test.VolumeID -ScheduleName "TEST-SF01-T2-T ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [New-SFSchedule], InvalidOperationException
    + FullyQualifiedErrorId : System.InvalidOperationException,SolidFire.Volume.New.NewSFSchedule
```

**This is without...**
 ```
New-SFSchedule -VolumeID $test.VolumeID -ScheduleName "TEST-SF01-T2-TEST99" -Frequency
Interval -TimeIntervalHours 6 -SnapshotName "Every6hFor24h-TEST-SF01-T2-TEST99"


Attributes       : {frequency}
HasError         : False
Hours            : 6
LastRunStatus    : Success
LastRunTimeStart :
Minutes          : 0
Monthdays        : {}
Paused           : False
Recurring        : True
RunNextInterval  : False
ScheduleID       : 3
ScheduleInfo     : SolidFire.Element.Api.ScheduleInfo
ScheduleName     : TEST-SF01-T2-TEST99
ScheduleType     : Snapshot
StartingDate     :
ToBeDeleted      : False
Weekdays         : {}

```

**Comment by ahaid** [comment]
This has been fixed in version 1.5 which is due to release in the next few weeks. If you would like an RC preview, I can send it to you via email. Send me a request at adam.haid@netapp.com. Otherwise, watch this repo for the next release.

**Comment by ahaid** [comment]
@rpmcdougall I was wrong. The problem is not fixed. We have logged the bug and will get the fix into the next release. In the meantime, a workaround is to include all the Retention* parameters like so:

    New-SFSchedule -VolumeID $test.VolumeID -ScheduleName "TEST-SF01-T2-TEST99" 
    -Frequency TimeInterval -TimeIntervalHours 6 -SnapshotName "Every6hFor24h-TEST-SF01-T2-TEST99" 
    -RetentionHours 24 -RetentionDays 0 -RetentionMinutes 0

The problem is rather dumb. When attempting to build the "00:00:00" retention time string, the underlying code gives up if it can't find a value for all the parameters. All our tests up to this point used all parameters and never tried leaving some off. No bueno! We will fix this so they default to 0 in the future. My apologies!

**Comment by Salmanius** [comment]
This has been fixed and will be included in the soon coming 1.6 release.

**Comment by arjun960** [comment]
This issue has been fixed, so closing the issue.
